airflow/contrib/hooks/spark_jdbc_script.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import argparse
+from pyspark.sql import SparkSession
+
+
+def set_common_options(spark_source,
+                       url='localhost:5432',
+                       jdbc_table='default.default',
+                       user='root',
+                       password='root',
+                       driver='driver'):
+
+    spark_source = spark_source \
+        .format('jdbc') \
+        .option('url', url) \
+        .option('dbtable', jdbc_table) \
+        .option('user', user) \
+        .option('password', password) \
+        .option('driver', driver)
+    return spark_source
+
+
+def spark_write_to_jdbc(spark, url, user, password, metastore_table, jdbc_table, driver,
+                        truncate, save_mode, batch_size, num_partitions,
+                        create_table_column_types):
+    writer = spark \
+        .table(metastore_table) \
+        .write \
+
+    # first set common options
+    writer = set_common_options(writer, url, jdbc_table, user, password, driver)
+
+    # now set write-specific options
+    if truncate:
+        writer = writer.option('truncate', truncate)
+    if batch_size:
+        writer = writer.option('batchsize', batch_size)
+    if num_partitions:
+        writer = writer.option('numPartitions', num_partitions)
+    if create_table_column_types:
+        writer = writer.option("createTableColumnTypes", create_table_column_types)
+
+    writer \
+        .save(mode=save_mode)
+
+
+def spark_read_from_jdbc(spark, url, user, password, metastore_table, jdbc_table, driver,
+                         save_mode, save_format, fetch_size, num_partitions,
+                         partition_column, lower_bound, upper_bound):
+    reader = spark.read
```

Comment:
Can we condense this a bit?
```
reader = spark.read
reader = set_common_options(reader, url, jdbc_table, user, password, driver)
```
to
```
reader = set_common_options(spark.read, url, jdbc_table, user, password, driver)
```
mutable variables make me cry in general, but I don't see a nicer way to do this in Python.